### ISSUE TITLE
Prop usage is not detected

### ISSUE THREAD

**Issue by budde377** [yannickcr/eslint-plugin-react#1107]
The following code:

```jsx
const React = require('react')
const someArray = require('./someArray')

const Test = props => <div>
    {someArray.map(l => <div
      key={l}>
        {props.property + props.property2}
      </div>)}
  </div>

Test.propTypes = {
  property: React.propTypes.string.isRequired,
  property2: React.propTypes.string.isRequired
}
```
raises error 

```
  12:13  error  'property' PropType is defined but prop is never used  react/no-unused-prop-types
````

with eslint-plugin-react version 6.10.0.

**Comment by ljharb** [comment]
What is `[].suppoerted`? that's not a property that would exist on arrays?

**Comment by budde377** [comment]
@ljharb That was a typo. I have corrected the example.

**Comment by sospedra** [comment]
Can confirm on v6.10.3 
It's because is inside the `map` function :/
